src/openklant/components/klantinteracties/migrations/0024_alter_betrokkene_bezoekadres_nummeraanduiding_id_and_more.py

```diff
@@ -0,0 +1,133 @@
+# Generated by Django 4.2.15 on 2025-01-09 09:22
+
+import json
+import logging
+import openklant.utils.validators
+from django.core.management import CommandError
+from django.db import migrations, models
+from django.db.models.functions import Length
+
+logger = logging.getLogger(__name__)
+
+
+def _log_records(records, model_name):
+    logger.info("--- START %s INFO ---", model_name)
+    logger.info(
+        "Found %s records for %s that have bezoekadres_nummeraanduiding_id "
+        "or correspondentieadres_nummeraanduiding_id length > 16 chars",
+        records.count(),
+        model_name,
+    )
+
+    for record in records:
+        logger.warning(
+            json.dumps(
+                {
+                    "model_name": model_name,
+                    "pk": record.pk,
+                    "uuid": str(record.uuid),
+                    "bezoekadres_nummeraanduiding_id": record.bezoekadres_nummeraanduiding_id,
+                    "correspondentieadres_nummeraanduiding_id": record.correspondentieadres_nummeraanduiding_id,
+                }
+            )
+        )
+
+    logger.info("--- END %s INFO ---", model_name)
+
+
+def _check_field_length(apps, schema_editor):
+
+    Betrokkene = apps.get_model("klantinteracties", "Betrokkene")
+    Partij = apps.get_model("klantinteracties", "Partij")
+
+    betrokkene_queryset = Betrokkene.objects.annotate(
+        length1=Length("bezoekadres_nummeraanduiding_id"),
+        length2=Length("correspondentieadres_nummeraanduiding_id"),
+    ).filter(models.Q(length1__gt=16) | models.Q(length2__gt=16))
```

Comment:
Same comment as in the other migration, this should include entries with <16 chars too I think